torch/testing/_internal/common_distributed.py

```diff
@@ -378,22 +379,28 @@ def _join_processes(self, fn):
                 # Sleep to avoid excessive busy polling.
                 time.sleep(0.1)
             elapsed_time = time.time() - start_time
-            if fn in self.skip_return_code_checks:
+            if fn in self.skip_process_error_code_checks:
+                # Don't check process exit codes, since these tests expect
+                # processes to crash/exit uncleanly. Also skips timeout
+                # checking since that is not needed due to the crash.
```

Comment:
I'm wondering why we need this separate handling here? When the process crashes, it doesn't exit with exit code 10 right? So it should be fine to keep the original logic that verified we didn't exit with `TEST_ERROR_EXIT_CODE`?